src/SFA.DAS.LevyTransferMatching.Infrastructure/Helpers/QuerystringHelper.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.Linq;
+using System.Collections.Generic;
+using System.Text;
+
+namespace SFA.DAS.LevyTransferMatching.Infrastructure.Helpers
+{
+    public class QuerystringHelper
+    {
+        public static string GetFormattedQuerystring(string prefix, string key, IEnumerable<string> values)
```

Comment:
I think there are a number of ways this could be improved.

Firstly, you might explore using the `QueryString` class, or failing that I have seen `NameValueCollection` used elsewhere. Eg. I've found this on the internet (not at all tested, and uses LINQ query comprehension syntax which we don't tend to use, but nonetheless as an example):

```        public static string ToQueryString(this NameValueCollection nvc)
        {
            var array = (
                from key in nvc.AllKeys
                from value in nvc.GetValues(key)
                select $"{HttpUtility.UrlEncode(key)}={HttpUtility.UrlEncode(value)}"
            ).ToArray();
            return "?" + string.Join("&", array);
        }
```

It's important to avoid string concatenation. I think an extension method or similar would also be a better way to do this.

Reply:
I have looked up and found a relatively good way (non LINQ) to build query string using NameValueCollection ... It could not be used out of box as it would not form query string with duplicates.